### PR TITLE
Add proxy configuration

### PR DESCRIPTION
**Pull request by coreykck**
Add getter and setter to Configuration Class and get the proxy to configure curl option.
You must pass the proxy as "host:port"

### CODE REVIEW

**Comment by napolux** [comment]
THIS IS JUST WHAT I NEED! PLEASE INTEGRATE!

**Comment by coreykck** [comment]
Tnx a lot napolux,
That's one small step for a man, a giant leap for mankind
Il 13/ago/2015 18:10, "Napolux" <notifications@github.com> ha scritto:

**Comment by reenl** [comment]
Need this :+1:

**Comment by wjr1985** [comment]
Hey @coreykck, @napolux, and @reenl,

Thanks for bringing this up! Just to update you, we are planning on implementing proxy configuration functionality into our client library. We'll keep you posted as to when it will be available.

File: lib/Braintree/Http.php [hunk 1]
```diff
@@ -131,6 +131,11 @@ public function _doUrlRequest($httpVerb, $url, $requestBody = null)
             curl_setopt($curl, CURLOPT_POSTFIELDS, $requestBody);
         }
 
+        if (!empty($this->_config->getProxy())) {
```
Comment (leon90) [inline lib/Braintree/Http.php]: Prior to PHP version 5.5 this will result in the following error:
**Can't use method return value in write context**

**Comment by coreykck** [comment]
Changed according to the comments. Tnx a lot

**Comment by wjr1985** [comment]
Hi @coreykck, @napolux, and @reenl,

We now support proxy configuration in our PHP library. To use it, you can specify the proxy in a similar way to how you configure your merchant ID, public key, and private key:

`Braintree_Configuration::proxyHost('https://localhost');`
`Braintree_Configuration::proxyPort('8089');`

If you want to specify the type of proxy (cURL's `CURLOPT_PROXYTYPE`), you can specify it by setting `Braintree_Configuration::proxyType()`. For example, if you're using a SOCKS5 proxy with a hostname, it would be `Braintree_Configuration::proxyType(CURLPROXY_SOCKS5_HOSTNAME);`

Official documentation is forthcoming, however this should be enough to get you started. If you run into any problems along the way feel free to open an issue.

**Comment by coreykck** [comment]
Tnx a lot bill.
Ciao
Corrado
Il 26/ott/2015 21:51, "Bill Rastello" <notifications@github.com> ha scritto:

**Comment by napolux** [comment]
Cool!

**Pull request closed without merging**